ma_desc.c

```diff
@@ -86,7 +86,7 @@ MADB_Desc *MADB_DescInit(MADB_Dbc *Dbc,enum enum_madb_desc_type DescType, my_boo
     MADB_FREE(Desc);
     Desc= NULL;
   }
-  if (isExternal)
+  if (Desc && isExternal)
```

Comment:
I guess would be cleaner to change both "Desc= NULL" to "return NULL", and remove Desc checks